data/src/main/java/com/microsoft/azure/kusto/data/KustoResultMapper.java

```diff
@@ -0,0 +1,132 @@
+package com.microsoft.azure.kusto.data;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BiConsumer;
+import java.util.function.Supplier;
+
+/**
+ * A class for mapping Kusto results to a list of pojos
+ * 
+ * @param <R>
+ *            pojo type returned by the mapping
+ */
+public class KustoResultMapper<R> {
+	
+	public static class Builder<R> {
+		final List<KustoResultColumnPopulator<R, ?, ?>> queryResultColumns = new ArrayList<>();
+		final Supplier<R> objConstructor;
+		
+		public Builder(Supplier<R> objConstructor) {
+			this.objConstructor = objConstructor;
+		}
+		
+		/**
+		 * Add a column by name. Using this function will cause the KustoResultMapper to lookup the column by name once per column per call to
+		 * {@link KustoResultMapper##extractList(KustoResultSetTable)} or {@link KustoResultMapper##extractSingle(KustoResultSetTable)}
```

Comment:
Javadoc validation is failing - I believe you want:
{@link KustoResultMapper#extractList(KustoResultSetTable)} or {@link KustoResultMapper#extractSingle(KustoResultSetTable)}

Reply:
okay, changed it.